File: packages/canvas/src-native/canvas-android/canvas/src/main/java/com/github/triniwiz/canvas/TNSImageBitmapResizeQuality.kt
package com.github.triniwiz.canvas

enum class TNSImageBitmapResizeQuality(val quality: String, val value: Int){
	Low("low", 0),
	Medium("medium", 1),
	High("high", 2),
	Pixelated("pixelated", 3);

	override fun toString(): String {
		return quality
	}

	fun toNative(): Int {
		return value
	}

	companion object {
		fun fromNative(value: Int): TNSImageBitmapResizeQuality? {
			return when (value) {
				0 -> Low
				1 -> Medium
				2 -> High
				3 -> Pixelated
				else -> null
			}
		}
	}
}

File: packages/canvas/src-native/canvas-ios/TNSImageBitmapResizeQuality.swift
//
//  TNSImageBitmapResizeQuality.swift
//  CanvasNative
//
//  Created by Osei Fortune on 11/04/2021.
//

import Foundation

@objc(TNSImageBitmapResizeQuality)
public enum TNSImageBitmapResizeQuality: Int, RawRepresentable {
    case Low
    case Medium
    case High
    case Pixelated
    public typealias RawValue = Int32
    
    public var rawValue: RawValue {
        switch self {
        case .Low:
            return 0
        case .Medium:
            return 1
        case .High:
            return 2
        case .Pixelated:
            return 3
        }
    }
    
    
    public init?(rawValue: RawValue) {
        switch rawValue {
        case 0:
            self = .Low
        case 1:
            self = .Medium
        case 2:
            self = .High
        case 3:
            self = .Pixelated
        default:
            return nil
        }
    }
    
    
    public init?(string: String) {
        switch string {
        case "low":
            self = .Low
        case "medium":
            self = .Medium
        case "high":
            self = .High
        case "pixelated":
            self = .Pixelated
        default:
            return nil
        }
    }
    
}
